Service/Generator.php

```diff
@@ -45,12 +46,18 @@ class Generator extends AbstractGenerator implements GeneratorInterface
      * @param EventDispatcherInterface $dispatcher
      * @param RouterInterface          $router
      * @param Cache|null               $cache
-     * @param integer|null             $cacheTtl
-     * @param integer|null             $itemsBySet
+     * @param int|null                 $cacheTtl
+     * @param int|null                 $itemsBySet
      */
-    public function __construct(EventDispatcherInterface $dispatcher, RouterInterface $router, Cache $cache = null, $cacheTtl = null, $itemsBySet = null)
-    {
+    public function __construct(
+        EventDispatcherInterface $dispatcher,
+        RouterInterface $router,
```

Comment:
Can we use `UrlGeneratorInterface` here (interface segregation)?

Reply:
Agreed.